api/src/test/java/com/exelaration/abstractmemery/controllers/MemeControllerTest.java

```diff
@@ -65,4 +68,37 @@ public void MemeControllerTest_WhenNullMemePosts_ExpectStatus400() throws Except
                 .content(new ObjectMapper().writeValueAsString(testMeme)))
         .andExpect(badRequest);
   }
+
+  @Test
+  public void getMemesForGallery_WhenMemeExists_ExpectStatus200andJSONreturn() throws Exception {
+    ResultMatcher ok = MockMvcResultMatchers.status().isOk();
+    String url = "/meme/";
+    ArrayList<String> expectedMemeData = new ArrayList<String>();
+    expectedMemeData.add("TestMemeData");
+
+    when(memeService.getMemes()).thenReturn(expectedMemeData);
+
+    mockMvc
+        .perform(MockMvcRequestBuilders.get(url))
+        .andExpect(ok)
+        .andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON_VALUE))
+        .andExpect(jsonPath("$", hasSize(1)))
+        .andExpect(jsonPath("$[0]", is("TestMemeData")));
+  }
+
+  @Test
+  public void getMemesForGallery_WhenMemeIsNull_ExpectStatus400andJSONreturn() throws Exception {
+    ResultMatcher badRequest = MockMvcResultMatchers.status().isBadRequest();
```

Comment:
I'm sorry to keep fixating on this part but now I have a question. Do we actually want it to be a bad request if there aren't any memes to be displayed? What if a user hasn't made any memes yet? Should it actually be a 200 and just be blank like the direct links ticket (when the image isn't found locally)?

Reply:
I agree with this.  If there's nothing to display, we just show an empty page, or text that says "No memes yet, upload one to get started."  Something like that.